src/_nebari/upgrade.py

```diff
@@ -507,6 +528,103 @@ def _version_specific_upgrade(
+        config["prevent_deploy"] = True
+
+        # Nebari version 2023.9.1 upgrades JupyterHub to 3.1.  CDS Dashboards are only compatible with
+        # JupyterHub versions 1.X and so will be removed during upgrade.
+        rich.print("\n ⚠️  Deprecation Warning ⚠️")
+        rich.print(
+            f"-> CDS dashboards are no longer supported in Nebari version [green]{self.version}[/green] and will be uninstalled."
+        )
+        if config.get("cdsdashboards"):
+            rich.print("-> Removing cdsdashboards from config file.")
+            del config["cdsdashboards"]
+
+        # Kubernetes version check.  Minimum Kubernetes version is 1.26
+        # JupyterHub Helm chart 2.0.0 (app version 3.0.0) requires K8S Version >=1.23. (reference: https://z2jh.jupyter.org/en/stable/)
+
+        provider = config["provider"]
+        provider_config_block = get_provider_config_block_name(provider)
+
+        # Get current Kubernetes version if available in config.
+        current_version = config.get(provider_config_block, {}).get(
+            "kubernetes_version", "NA"
+        )
+
+        # Convert to decimal prefix
+        if provider in ["aws", "azure", "gcp", "do"]:
+            current_version = get_k8s_version_prefix(current_version)
+
+        # Try to convert known Kubernetes versions to float.
+        if not current_version == "NA":
+            try:
+                current_version = float(current_version)
+            except TypeError:
+                current_version = "NA"
+
+        # Handle checks for when Kubernetes version should be detectable
+        if provider in ["aws", "azure", "gcp", "do"]:
+            # Kubernetes version not found in provider block
+            if current_version == "NA":
+                rich.print("\n ⚠️  Warning ⚠️")
+                rich.print(
+                    f"-> Unable to detect Kubernetes version for provider {provider}.  Nebari version [green]{self.version}[/green] requires Kubernetes version 1.26.  Please confirm your Kubernetes version is configured before upgrading."
+                )
+
+            # Kubernetes version less than required minimum
+            if isinstance(current_version, float) and current_version < 1.26:
+                rich.print("\n ⚠️  Warning ⚠️")
+                rich.print(
+                    f"-> Nebari version [green]{self.version}[/green] requires Kubernetes version 1.26.  Your configured Kubernetes version is [red]{current_version}[/red]. {UPGRADE_KUBERNETES_MESSAGE}"
+                )
+                version_diff = round(1.26 - current_version, 2)
+                if version_diff > 0.01:
+                    rich.print(
+                        "-> The Kubernetes version is multiple minor versions behind the minimum required version. You will need to perform the upgrade one minor version at a time.  For example, if your current version is 1.24, you will need to upgrade to 1.25, and then 1.26."
```

Comment:
> You will need to perform the upgrade one minor version at a time

What was the reason for this? Is there a doc link somewhere?

Reply:
I wonder if this is a requirement enforced by the cloud providers because I see this same warning on their docs ([GKE for example](https://cloud.google.com/kubernetes-engine/docs/how-to/upgrading-a-cluster#upgrade_cp)).

Reply:
I see, thanks for clarification.